doc/rfcs/login_block_sliding_window.md

```diff
@@ -0,0 +1,66 @@
+# Login Sliding-Window Blocks
+
+## Overview and motivation
+`ms-users` service supports some IP and username blocks. The Current version uses block on TTL'ed counter in the Redis database,
+but this brings some strong limits for users. Service monitors count of login attempts from IP address and IP + Username,
+if one of the counters becomes greater than specified config parameter, all requests from this IP or IP+username blocked
+for TTL. For better user experience introducing a new feature
+
+## Sliding window Block
+Solution based on `inredis-ratelimiter`(https://github.com/3hedgehogs/inredis-ratelimiter) with some additions and fixes, replaces the standard block counter(int key) +TTL.
+
+Sliding-window provides better control on operations/requests processed in some period.
+
+## utils/slidingwindow/limiter.js
+Exports Redis script function wrappers for scripts `sWindowReserve`, `sWindowCheck`, `sWindowCancel`.
+#### reserve(redis, key, interval, limit)
+Wraps `sWindowReserve`. Returns `{ usage: int, token: string/microseconds, reset: int }`
+#### check(redis, key, interval, limit)
+Wraps `sWindowCheck`. Returns `{ usage: int, reset: int}`
+#### cancel(redis, key, token)
+Wraps `sWindowCancel`. Returns count of deleted spans/tokens.
+
+##### Example
+```javascript
+const limiter = require('util/slidingwindow/limiter');
+// Try to allocate span in sliding window now()-interval seconds with max 10 slots/tokens
+const {usage, token, reset} = await limiter.reserve(redis, "dataKey", 60 * 60, 10);
+if (!token) {
+  // Unable to reserve span/token. Limit exceeded.
+  console.log(`Wait for ${reset} seconds`);
+}
+// Token/span assigned
+// We can proceed our work
+...
+// Free used span/token.
+await limiter.cancel(token);
+```
+
+### sWindowReserve
+Script uses hires timestamp(microseconds) as scores in sorted SET to avoid some timestamp collision conditions.
+Deletes outdated inside of the current sliding `interval` entries in provided sorted SET.
+Gets count of members in SET, if members count is greater than `limit` returns `reset` milliseconds(interval to next execution available).
+Assigns TTL for `key` value `interval`, if `interval` is 0 TTL not set.
+When `interval` is 0 value of `reset` may become negative, in this case, `reset` return 0
+### sWindowCheck
+Used to check current usage count and projected reset interval for provided `key`. When executed automatically cleans
```

Comment:
Script title sounds like it doesn't change db state, maybe smth like `sWindowRefresh`?